tests/integration/transactions/test_set_oracle.py

```diff
@@ -0,0 +1,52 @@
+import random
+import time
+
+from tests.integration.integration_test_case import IntegrationTestCase
+from tests.integration.it_utils import (
+    sign_and_reliable_submission_async,
+    test_async_and_sync,
+)
+from tests.integration.reusable_values import WALLET
+from xrpl.models import AccountObjects, AccountObjectType, OracleSet
+from xrpl.models.response import ResponseStatus
+from xrpl.models.transactions.oracle_set import PriceData
+from xrpl.utils import str_to_hex
+
+_PROVIDER = str_to_hex("provider")
+_ASSET_CLASS = str_to_hex("currency")
+
+
+class TestSetOracle(IntegrationTestCase):
+    @test_async_and_sync(globals())
+    async def test_all_fields(self, client):
+        tx = OracleSet(
+            account=WALLET.address,
+            # if oracle_document_id is not modified, the (sync, async) +
+            # (json, websocket) combination of integration tests will update the same
+            # oracle object using identical "LastUpdateTime". Updates to an oracle must
+            # be more recent than its previous LastUpdateTime
+            oracle_document_id=random.randint(100, 300),
```

Comment:
Instead of having a random number here, why not increment a global variable? It'll be easier to track that way.

Reply:
I tried your suggestion, but there is a problem.

Identical version of the source code is parsed for `json` and `websocket` clients. Hence both of these tests get the same oracle_document_id values.

This behavior is seen for both `sync` and `async` tests. I'm not sure how we can work around that. I think we can tweak `tests/integration/it_utils.py:test_async_and_sync` to achieve this effect, but it's not obvious to me right now.

here is my attempt: https://github.com/XRPLF/xrpl-py/commit/bf193fb38545d7fae14092dbb46991f5d745d29f

Reply:
Yes, that's basically what I'm suggesting.

Reply:
yeah, but I don't know how we can get it to work.

anyway, this is a moot point, this file has been removed

Reply:
Here's a patch that can give you a unique value for each client: https://github.com/XRPLF/xrpl-py/commit/d08ba7fc00d8d4eaaa53a270d91f345ab1369427

Reply:
thanks @mvadari  this is a very interesting solution!

haha, the `self` is getting replced by the decorator :)

I've pulled this commit into the latest tip of this branch